foam_controlwrapper/foam_controlwrapper.py

```diff
@@ -174,42 +97,179 @@ def run(self):
+            information.
 
-    def add_lattice(self, lattice):
-        raise NotImplementedError()
+        Raises
+        ------
+        Exception if mesh already exists
 
-    def add_mesh(self, mesh):
-        raise NotImplementedError()
+        """
 
-    def delete_lattice(self, name):
-        raise NotImplementedError()
+        if mesh.name in self._meshes:
+            raise ValueError('Mesh \'{}\` already exists'.format(mesh.name))
+        else:
+            self._meshes[mesh.name] = FoamMesh(mesh.name, mesh)
+            return self._meshes[mesh.name]
 
     def delete_mesh(self, name):
-        raise NotImplementedError()
+        """Delete mesh from the OpenFoam modeling engine.
 
-    def get_lattice(self, name):
-        raise NotImplementedError()
+        Parameters
+        ----------
+        name : str
+            name of the mesh to be deleted.
+
+
+        Raises
+        ------
+        Exception if mesh not found
+
+        """
+
+        if name not in self._meshes:
+            raise ValueError('Mesh \'{}\` does not exists'.format(name))
+        else:
+            simphonyfoaminterface.deleteMesh(name)
+            del self._meshes[name]
 
     def get_mesh(self, name):
-        raise NotImplementedError()
+        """Get a mesh.
 
-    def iter_lattices(self, names=None):
-        raise NotImplementedError()
+        The returned mesh can be used to query and update the state of the
+        mesh inside the OpenFoam modeling engine.
+
+        Parameters
+        ----------
+        name : str
+            name of the mesh to be retrieved.
+
+        Returns
+        -------
+        FoamMesh
+
+        Raises
+        ------
+        Exception if mesh not found
+
+        """
+
+        if name in self._meshes:
+            return self._meshes[name]
+        else:
+            raise ValueError(
+                'Mesh \'{}\` does not exist'.format(name))
 
     def iter_meshes(self, names=None):
-        raise NotImplementedError()
+        """Returns an iterator over a subset or all of the meshes.
+
+        Parameters
+        ----------
+        names : list of str
+            names of specific meshes to be iterated over.
+            If names is not given, then all meshes will
+            be iterated over.
+
+        Returns
+        ----------
+        Iterator over a subset or all of the meshes
+
+        Raises
+        ------
+        Exception if some mesh fron mesh names list not found
+
+        """
+
+        if names is None:
+            for name in self._meshes:
+                yield self._meshes[name]
+        else:
+            for name in names:
+                if name in self._meshes:
+                    yield self._meshes[name]
+                else:
+                    raise ValueError(
+                        'Mesh \'{}\` does not exist'.format(
+                            name))
+
+    def read_foammesh(self, name, path):
```

Comment:
*API suggestion*: The ``read_foammesh`` and ``write_foammesh`` could be defined outside the modeling engine class as module level utility functions.